langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/common/AzureOpenAiChatModelIT.java

```diff
@@ -0,0 +1,89 @@
+package dev.langchain4j.model.azure.common;
+
+import dev.langchain4j.model.azure.AzureOpenAiChatModel;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.chat.common.AbstractChatModelIT;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+
+class AzureOpenAiChatModelIT extends AbstractChatModelIT {
+
+    // TODO https://github.com/langchain4j/langchain4j/issues/2219
+
+    static final AzureOpenAiChatModel AZURE_OPEN_AI_CHAT_MODEL = AzureOpenAiChatModel.builder()
+            .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
+            .apiKey(System.getenv("AZURE_OPENAI_KEY"))
+            .deploymentName("gpt-4o-mini")
+            .build();
+
+    @Override
+    protected List<ChatLanguageModel> models() {
+        return List.of(
+                AZURE_OPEN_AI_CHAT_MODEL
+                // TODO add more model configs, see OpenAiChatModelIT
+        );
+    }
+
+    @Override
+    @Disabled
+    @ParameterizedTest
+    @MethodSource("modelsSupportingImageInputs")
+    protected void should_accept_single_image_as_public_URL(ChatLanguageModel model) {
+        // TODO fix
```

Comment:
Hi @jdubois, something strange is going on when images are involved, the test seemingly hangs forever. I would appreciate any help (investigation/fix) with this 🙏

Reply:
At least I can reproduce the test hanging :-D

Reply:
@langchain4j can you add  `.logRequestsAndResponses(true)` when creating `AZURE_OPEN_AI_CHAT_MODEL` so you can check the logs?
On my side I immediately hit a `429` telling me `Please retry after 86400 seconds`, so I'm out of quota for the day!! Then that should trigger an error, not wait forever.

Reply:
Oh, that is interesting... Thank you for looking into it! We really need to fix error handling in Azure OpenAI module

Reply:
I've debuged it, it hangs on the line `ChatCompletions chatCompletions = client.getChatCompletions(deploymentName, options);` line inside `AzureOpenAiChatModel.generate()`. Seems like an issue in the Azure OpenAI SDK